### PR TITLE
Feature/cache wildcard 0.2.2

### PR DESCRIPTION
**Pull request by drc-creare**
- Changes `CacheCtrl.rem()` so that wildcard matches for inputs are specified with '*' instead of None, As sometimes you specifically want to match just `None` (e.g. `coordinates=None`)
- Fixes two failing tests for cache in `test_node.py`
- Some documentation additions/updates
- Adds some type checking for function arguments for CacheCtrl methods.

### CODE REVIEW

File: podpac/core/cache/cache.py [hunk 1]
```diff
@@ -85,27 +151,43 @@ def get(self, node, key, coordinates=None, mode=None):
...
+        assert not isinstance(key, CacheWildCard)
         mode = self._determine_mode(mode)
         for c in self._cache_stores:
             if c.cache_modes_matches(set([mode])):
                 if c.has(node=node, key=key, coordinates=coordinates):
                     return c.get(node=node, key=key, coordinates=coordinates)
         raise CacheException("Requested data is not in any cache stores.")
 
-    def rem(self, node=None, key=None, coordinates=None, mode=None):
+    def rem(self, node='*', key='*', coordinates='*', mode=None):
```
[the remark this answers is not in the document]
Comment (drc-creare) [reply podpac/core/cache/cache.py]: Yeah, good call. But there is no good default for node and key. I'll make those non-optional.

File: podpac/core/node.py [hunk 2]
```diff
@@ -512,16 +512,16 @@ def has_cache(self, key, coordinates=None):
             return False
         return self.cache_ctrl.has(self, key, coordinates=coordinates)
 
-    def rem_cache(self, key=None, coordinates=None, mode=None, all_cache=False):
+    def rem_cache(self, key='*', coordinates='*', mode=None, all_cache=False):
```
[the remark this answers is not in the document]
Comment (drc-creare) [reply podpac/core/node.py]: Good call, but will have to make key a non-optional parameters (no default), `None` will actually throw an error.

File: podpac/core/cache/cache.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmilloy) [reply podpac/core/cache/cache.py]: So you can't cache something without a node?

[the remark this answers is not in the document]
Comment (drc-creare) [reply podpac/core/cache/cache.py]: Yes, only node's can cache things.

File: podpac/core/node.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mpu-creare) [reply podpac/core/node.py]: That sounds good.

**Pull request merged**